pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -148,116 +198,31 @@ func (r *CloudEntitySelectorReconciler) processCreateOrUpdate(selector *cloudv1a
-
-	if pollerScope, exists := r.accPollers[*selectorNamespacedName]; exists {
-		r.Log.Info("poller exists", "selector", selectorNamespacedName)
-		return pollerScope, exists
-	}
-
-	accountNamespacedName := &types.NamespacedName{
-		Namespace: selector.Namespace,
-		Name:      selector.Spec.AccountName,
+	tempAccountNamespacedName := accountNamespacedName
+	delete(r.selectorToAccountMap, *selectorNamespacedName)
+	accPoller, _ := GetAccountPoller(&tempAccountNamespacedName)
+	if accPoller == nil {
+		r.Log.Info("account poller is already deleted", "accountNamespacedName", tempAccountNamespacedName.String())
+		return nil
 	}
-	account := &cloudv1alpha1.CloudProviderAccount{}
-	_ = r.Get(context.TODO(), *accountNamespacedName, account)
-	accountCloudType, err := account.GetAccountProviderType()
+	cloudInterface, err := cloudprovider.GetCloudInterface(common.ProviderType(accPoller.cloudType))
 	if err != nil {
-		return nil, false
-	}
-	poller := &accountPoller{
-		Client:            r.Client,
-		scheme:            r.Scheme,
-		log:               r.Log,
-		pollIntvInSeconds: *account.Spec.PollIntervalInSeconds,
-		cloudType:         accountCloudType,
-		namespacedName:    accountNamespacedName,
-		selector:          selector.DeepCopy(),
-		ch:                make(chan struct{}),
+		return err
 	}
-	poller.vmSelector = cache.NewIndexer(
-		func(obj interface{}) (string, error) {
-			m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-			// Create a unique key for each VirtualMachineSelector.
-			return fmt.Sprintf("%v-%v-%v", m.Agented, m.VpcMatch, m.VMMatch), nil
-		},
-		cache.Indexers{
-			virtualMachineSelectorMatchIndexerByID: func(obj interface{}) ([]string, error) {
-				m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-				if len(m.VMMatch) == 0 {
-					return nil, nil
-				}
-				var match []string
-				for _, vmMatch := range m.VMMatch {
-					if len(vmMatch.MatchID) > 0 {
-						match = append(match, strings.ToLower(vmMatch.MatchID))
-					}
-				}
-				return match, nil
-			},
-			virtualMachineSelectorMatchIndexerByName: func(obj interface{}) ([]string, error) {
-				m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-				if len(m.VMMatch) == 0 {
-					return nil, nil
-				}
-				var match []string
-				for _, vmMatch := range m.VMMatch {
-					if len(vmMatch.MatchName) > 0 {
-						match = append(match, strings.ToLower(vmMatch.MatchName))
-					}
-				}
-				return match, nil
-			},
-			virtualMachineSelectorMatchIndexerByVPC: func(obj interface{}) ([]string, error) {
-				m := obj.(*cloudv1alpha1.VirtualMachineSelector)
-				if m.VpcMatch != nil && len(m.VpcMatch.MatchID) > 0 {
-					return []string{strings.ToLower(m.VpcMatch.MatchID)}, nil
-				}
-				return nil, nil
-			},
-		})
 
-	r.accPollers[*selectorNamespacedName] = poller
-
-	r.Log.Info("poller will be created", "selector", selectorNamespacedName)
-	return poller, false
-}
+	cloudInterface.RemoveAccountResourcesSelector(&tempAccountNamespacedName, selectorNamespacedName.Name)
```

Comment:
With this change, I'm assuming that the order of deletion of CPA/CES, does not matter and it will clear the filters/cached state.

Reply:
Right